testcases/kernel/security/stack_clash/stack_clash.c

```diff
@@ -0,0 +1,36 @@
+/* File from https://bugzilla.novell.com/show_bug.cgi?id=CVE-2017-1000364 */
+
+#include <sys/mman.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <alloca.h>
+#include <signal.h>
+#include <stdlib.h>
+
+#include "tst_test.h"
+
+#define FNAME "test"
+
+void stack_clash_test(void)
+{
+	int status;
+	pid_t pid;
+
+	pid = SAFE_FORK();
+	if (!pid) {
+		SAFE_EXECLP("stack_clash_child", "stack_clash_child", NULL);
+	}
+
+	SAFE_WAITPID(pid, &status, 0);
+
+	if (WEXITSTATUS(status))
+		tst_res(TFAIL, "stack is too clsoe to the mmaped area");
```

Comment:
This will show same error message no matter where child fails. I'd add "WIFEXTED() &&"